source/slang/syntax-base-defs.h

```diff
@@ -195,6 +195,40 @@ SYNTAX_CLASS(ThisTypeSubstitution, Substitutions)
         return sourceType->GetHashCode();
     }
     )
+    END_SYNTAX_CLASS()
+
+    SYNTAX_CLASS(GlobalGenericParamSubstitution, Substitutions)
+    // the __generic_param decl to be substituted
+    DECL_FIELD(GlobalGenericParamDecl*, paramDecl)
+    // the actual type to substitute in
+    SYNTAX_FIELD(RefPtr<Val>, actualType)
+    // The witness tables for each interface this actual type implements
+    typedef List<KeyValuePair<RefPtr<Type>, RefPtr<Val>>> WitnessTableLookupTable;
```

Comment:
I think this line needs to be in a `RAW(...)` or else you might accidentally make that a global `typedef`.